lightning/src/routing/scoring.rs

```diff
@@ -678,25 +795,107 @@ impl<L: Deref<Target = u64>, T: Time, U: Deref<Target = T>> DirectedChannelLiqui
 			} else {
 				let negative_log10_times_2048 =
 					approx::negative_log10_times_2048(numerator, denominator);
-				self.combined_penalty_msat(amount_msat, negative_log10_times_2048, params)
+				Self::combined_penalty_msat(amount_msat, negative_log10_times_2048,
+					params.liquidity_penalty_multiplier_msat,
+					params.liquidity_penalty_amount_multiplier_msat)
+			}
+		};
+
+		if params.historical_liquidity_penalty_multiplier_msat != 0 ||
+		   params.historical_liquidity_penalty_amount_multiplier_msat != 0 {
+			// If historical penalties are enabled, calculate the penalty by walking the set of
+			// historical liquidity bucket (min, max) combinations (where the min is less than the
+			// max) and, for each, calculate the probability of success given our payment amount,
+			// then total the weighted average probability of success.
+			//
+			// We use a sliding scale to decide which point within a given bucket will be compared
+			// to the amount being sent - the amount being sent is compared to the lower-bound of
```

Comment:
Wait, as I read it now it says that we compare the `amount` to the lower-bound in case of the left-most bucket and to the upper-bound in the right-most bucket. But what happens for all the buckets in between?